### ISSUE TITLE
updateProduct > mediums and purposes field value updated even if fields are not provided in the input

### ISSUE THREAD

**Issue by iPiyushPatel** [SeedCompany/cord-api-v3#1147]
**Describe the bug**
updateProduct > `mediums` and `purposes` field value updated even if fields are not provided in the input

*Steps to Reproduce*
On Graphql playground or Postman - create an API request as below snap & attached notepad file and execute the request

**Expected Result**
 `mediums` and `purposes` field value should not be updated

**Actual Result**
`mediums` and `purposes` field value updated even if fields are not provided in the input
See video -  https://drive.google.com/file/d/1_Cp1Zl-tGPvypSlg1sCb6HZnPxHvrXfm/view?usp=sharing 

**GraphQL Query(updateproduct with input ):**
![image](https://user-images.githubusercontent.com/5212585/92593069-26fc5300-f2be-11ea-8d36-bd0498b232e2.png)

**GraphQL Query(updateproduct without input ):**
![image](https://user-images.githubusercontent.com/5212585/92593257-72166600-f2be-11ea-9e7c-a56167fc9c9b.png)

**GraphQL response**
```
 "data": {
        "updateProduct": {
            "product": {
                "id": "6uMCLWH6x",
                "createdAt": "2020-09-09T16:35:10.652+05:30",
                "approach": "OralStories",
                "legacyType": "BibleStories",
                "mediums": {
                    "value": [
                        "Web"
                    ],
                    "canEdit": true,
                    "canRead": true
                },
                "purposes": {
                    "value": [
                        "Discipleship"
                    ],
                    "canEdit": true,
                    "canRead": true
                },
                "methodology": {
                    "value": "BibleStorying",
                    "canEdit": true,
                    "canRead": true
                },
```

**Query and Input File**
https://docs.google.com/document/d/1vfw9igBGEN-9xMPmrCWJNbEeQS-OzH0n_usTKOfAKe0/edit?usp=sharing

CordAPIV3# ac33147f

**Comment by zkhin** [comment]
@iPiyushPatel Can you reverify this issue?
I tried to update the `product` without `mediums` and `purposes` fields but they are still the same.

**Comment by iPiyushPatel** [comment]
@zkhin @reneepe @michaelmarshall 
Verified on the latest master af9851eb and working fine now.

Note: As the `rootDirectory` field in query output is causing issue on the latest master af9851eb , I've tested by removing it from query output and It's working as expected.

**Comment by tjgersho** [comment]
I assume you mean https://github.com/SeedCompany/cord-api-v3/issues/1174

**Comment by reneepe** [comment]
Bug fixed with #1174? Closing ticket.
